cirq/study/resolver.py

```diff
@@ -42,7 +42,7 @@ class ParamResolver:
     A ParamResolver is an object that can be used to assign values for these
     variables.
 
-    ParamResolvers are hashable.
+    ParamResolvers are hashable; their param_dict must not be mutated.
```

Comment:
Out of scope of this PR: I wonder if it's easy to make it private

Reply:
I tried making it a `@property`, but that apparently doesn't stop users from mutating its contents :(